private/buf/cmd/buf/command/breaking/breaking.go

```diff
@@ -400,3 +434,30 @@ func getExternalPathsForImages(imageConfigs []bufwire.ImageConfig, excludeImport
 	}
 	return stringutil.MapToSlice(externalPaths), nil
 }
+
+func mergeImageConfigs(
+	ctx context.Context,
+	imageConfigs []bufwire.ImageConfig,
+	readBucket storage.ReadBucket,
+	inputConfig string,
+) (bufwire.ImageConfig, error) {
+	config, err := bufconfig.ReadConfigOS(
```

Comment:
This all needs more explanation as to what is going on, I'm a bit confused here

Reply:
Yea I don't think this is right. This looks like we're reading a single configuration file from the OS and using it for the  entire workspace - this would violate the property we need to maintain that's described [here](https://github.com/bufbuild/buf/issues/567#issuecomment-936156567).

Reply:
This is the first proposed approach where we try to emulate
```bash
buf breaking --against against.bin $(buf build .) 
```
This is similar to the behaviour explained [here](https://github.com/bufbuild/buf/issues/567#issuecomment-998149291).

Here we are reading from `--against-config` and `--config` if not found will revert to the OS.